torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -0,0 +1,118 @@
+#include "torch_xla/csrc/ops/dynamic_ir.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+static const torch::lazy::DimensionNode* DimCast(torch::lazy::Output output) {
+  return dynamic_cast<const torch::lazy::DimensionNode*>(output.node);
+}
+
+namespace torch_xla {
+
+SizeNode::SizeNode(torch::lazy::Value input, size_t dim)
+    : XlaNode(torch::lazy::OpKind{c10::Symbol::fromQualString("aten::size")},
+              {input}, xla::ShapeUtil::MakeShape(xla::S32, {}), 1,
+              torch::lazy::MHash(dim)),
+      dim_(dim){};
+
+XlaOpVector SizeNode::Lower(LoweringContext* loctx) const {
+  auto input = loctx->GetOutputOp(operand(0));
+  return ReturnOp(xla::GetDimensionSize(input, this->dim_), loctx);
+}
+
+int64_t SizeNode::getStaticValue() const {
+  return dynamic_cast<const XlaNode*>(operand(0).node)->shape(0).size(dim_);
```

Comment:
`shape` will return a `torch::lazy::shape` I think? If that's case why not just leave it as `torch::lazy::Node`, `shape()` also existed on upstream base class.

Reply:
Well, either way works. 

This is mainly a design choice that provides explicit clarity that `SizeNode` delivers a staticValue, implying the value is not necessarily the true shape.

Reply:
I guess my question is more of why not use `operand(0).node->shape(0)` directly? What's the type of `operand(0).node`?

Reply:
Thanks for the clarification. Yes, I call `shape()` directly - change coming up now. The reason it was initially set like this is due to a legacy reason - we didn't use to have multiple inheritance for `SizeNode` initially; now that we do, we can do what you suggested.